### PR TITLE
[FM v0.4.0]-- Allow key-value Environment variable passthrough

### PR DESCRIPTION
**Pull request by Huanli-Meng**
### Motivation
The [code PR #424](https://github.com/streamnative/function-mesh/pull/424) introduce `.spec.env` that gets combined with `.spec.secretsMap` to form the environment variables for the statefulset/pods deployed by this.

Therefore, update the doc accordingly.

### Modification
- Add the `env` option and its descriptions in Function/Source/Sink CRD configuration files.
- Change the Pod Specifications table form MD format to HTML format as we need to add code in a table cell.
- Affected release: latest (FM v0.4.0)

### CODE REVIEW

File: docs/connectors/io-crd-config/sink-crd-config.md [hunk 1]
```diff
@@ -171,3 +171,4 @@ Function Mesh supports customizing the Pod running Pulsar connectors. This table
...
 | `autoScalingMetrics` | Specify how to scale based on customized metrics defined in connectors. For details, see [MetricSpec v2beta2 autoscaling](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.21/#metricspec-v2beta2-autoscaling). |
 | `autoScalingBehavior` | Configure the scaling behavior of the target in both up and down directions (`scaleUp` and `scaleDown` fields respectively). If not specified, the default Kubernetes scaling behaviors are adopted. For details, see [HorizontalPodAutoscalerBehavior v2beta2 autoscaling](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.21/#horizontalpodautoscalerbehavior-v2beta2-autoscaling). |
+| `env` | Specify the environment variables to expose on the containers. It is a key/value map. You can either specify a particular value for the environment variable or use the `valueFrom` option to specify the [source](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.21/#envvarsource-v1-core) for the environment variable's value. |
```
Comment (jdmaguire) [inline docs/connectors/io-crd-config/sink-crd-config.md:174]: Now sure if this is clear.

It is a list like:

```
- name: "EXAMPLE_1"
   value: "VALUE_1"
- name: "EXAMPLE_2"
  value: "VALUE_2" 
 ```

Comment (Huanli-Meng) [reply docs/connectors/io-crd-config/sink-crd-config.md]: @jdmaguire Thanks for your comment. I add a YAML code example in the table cell. It looks like this. Could you please help whether it is OK or not? Thanks.
image

**Review by jiangpengcheng — APPROVED** [review]

File: docs/connectors/io-crd-config/source-crd-config.md [hunk 2]
```diff
@@ -147,20 +147,90 @@ In Function Mesh, the Pulsar cluster is defined through a ConfigMap. Pods can co
...
+    <td><code>volumes</code></td>
+    <td>It is a list of volumes that can be mounted by containers belonging to a Pod.</td>
+  </tr>
+  <tr>
+    <td><code>imagePullSecrets</code></td>
+    <td>It is an optional list of references to secrets in the same namespace for pulling any of the images used by a Pod.</td>
+  </tr>
+  <tr>
+    <td><code>serviceAccountName</code></td>
+    <td>Specify the name of the service account which is used to run Pulsar Functions or connectors in the Function Mesh Worker service.</td>
```
Comment (teachwrite) [inline docs/connectors/io-crd-config/source-crd-config.md:196]:
```suggestion
    <td>Specify the name of the service account that is used to run Pulsar Functions or connectors in the Function Mesh Worker service.</td>
```

File: docs/functions/function-crd.md [hunk 3]
```diff
@@ -189,20 +189,90 @@ In Function Mesh, the Pulsar cluster is defined through a ConfigMap. Pods can co
...
+    <td><code>securityContext</code></td>
+    <td>Specify the security context for a Pod.</td>
+  </tr>
+  <tr>
+    <td><code>terminationGracePeriodSeconds</code></td>
+    <td>It is the amount of time that Kubernetes gives for a Pod before terminating it.</td>
+  </tr>
+  <tr>
+    <td><code>volumes</code></td>
+    <td>It is a list of volumes that can be mounted by containers belonging to a Pod.</td>
```
Comment (teachwrite) [inline docs/functions/function-crd.md:230]:
```suggestion
    <td>A list of volumes that can be mounted by containers belonging to a Pod.</td>
```

**Review by teachwrite — APPROVED** [review]
Just a few minor changes in the files (same changes across all of the  files).

**Review by jiangpengcheng — APPROVED** [review]

**Comment by freeznet** [comment]
@Huanli-Meng @jiangpengcheng I think we should add the env to mesh worker service configs as well, like https://functionmesh.io/docs/next/function-mesh-worker/reference/customizable-option and https://functionmesh.io/docs/next/function-mesh-worker/reference/general-option. I have created an issue https://github.com/streamnative/function-mesh-website/issues/175.

**Comment by Huanli-Meng** [comment]
> @Huanli-Meng @jiangpengcheng I think we should add the env to mesh worker service configs as well, like https://functionmesh.io/docs/next/function-mesh-worker/reference/customizable-option and https://functionmesh.io/docs/next/function-mesh-worker/reference/general-option. I have created an issue #175.

Thanks, @freeznet I will update the doc tomorrow.

**Pull request merged**